lib/util.js

```diff
@@ -137,34 +135,6 @@ function strEscape(str) {
   return `'${result}'`;
 }
 
-// Escape control characters and single quotes.
-// Note: for performance reasons this is not combined with strEscape
-function keyEscape(str) {
-  if (str.length < 5000 && !keyEscapeSequencesRegExp.test(str))
-    return `'${str}'`;
-  if (str.length > 100)
-    return `'${str.replace(keyEscapeSequencesReplacer, escapeFn)}'`;
```

Comment:
Not specific to this PR but to the code:

@bmeurer in my tests `RegExp.p.test` performs better than `String.p.replace`. Should those not be on par in case no match is found? And with small strings it is (last tested with 6.1) still better to use `String.p.charCodeAt` instead of the RegExp. Is there any chance to improve the RegExp so it would match the performance of `charCodeAt`?

Reply:
> @bmeurer in my tests RegExp.p.test performs better than String.p.replace. Should those not be on par in case no match is found?

`test` is a much simpler operation than `@@replace` (see the spec at [0] and [1]). It's fairly easy to implement `test` efficiently; but in my experience `@@replace` is another story, see V8's `@@replace` dispatch logic at [2] to feel our pain.

In this particular case, we seem to reach the `ReplaceGlobalCallableFastPath`. There's definitely a couple of things we could do to improve here, like remove one (or both) runtime calls. Could you open a bug at crbug.com/v8/new?

[0] https://tc39.github.io/ecma262/#sec-regexp.prototype.test
[1] https://tc39.github.io/ecma262/#sec-regexp.prototype-@@replace
[2] https://cs.chromium.org/chromium/src/v8/src/builtins/builtins-regexp-gen.cc?l=3058&rcl=2eea37273b30caa5cedf3a5c8d656860bc60320b

> And with small strings it is (last tested with 6.1) still better to use String.p.charCodeAt instead of the RegExp. Is there any chance to improve the RegExp so it would match the performance of charCodeAt?

I assume you mean `charCodeAt` vs. `test`? It'll be hard to beat, since `charCodeAt` is completely inlined by the optimizing compiler. I suppose one possible step we could take would be to eliminate the call overhead from RegExp builtins, but I wouldn't expect performance to improve by alot.

Could you share the benchmark you used to measure this?

Reply:
> Could you open a bug at crbug.com/v8/new?

Went ahead and created https://crbug.com/v8/7081.